lib/Doctrine/ORM/Query/ParameterTypeInferer.php

```diff
@@ -53,7 +53,7 @@ public static function inferType($value)
             return Type::BOOLEAN;
         }
 
-        if ($value instanceof \DateTime) {
+        if ($value instanceof \DateTime || $value instanceof \DateTimeImmutable) {
```

Comment:
@janlanger I would still be in favor of checking the interface rather than ``\DateTimeImmutable`` to support PHP 5.5+